libs/execution_context/ebpf_link.c

```diff
@@ -112,7 +114,10 @@ ebpf_link_attach_program(ebpf_link_t* link, ebpf_program_t* program)
     }
 
     link->program = program;
-    ebpf_object_acquire_reference((ebpf_object_t*)program);
+    ebpf_program_add_link_to_list(program, link);
+
+    // Take "attach" reference which will be released when detach is called.
```

Comment:
may want to update object->reference_count to be a union of array of typed counts and a total count. For the "typed" counts the array index will give the ref type. 0 - handle, 1 - attach etc.